### ISSUE TITLE
_source filtering input validation & behavior

### ISSUE THREAD

**Issue by n0othing** [elastic/elasticsearch#19008]
[_source filtering](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-source-filtering.html) returns the full contents of the `_source` when supplied with a number (as opposed to a string). It would be better to return nothing and/or not be valid + throw an error.

Add a document:

```
PUT myindex/mytype/1
{
  "field1": "Hello World!"
}
```

Pass in a number:

```
GET myindex/_search
{
  "_source": 1
}
```

The result:

```
{
  "took": 1,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "failed": 0
  },
  "hits": {
    "total": 1,
    "max_score": 1,
    "hits": [
      {
        "_index": "myindex",
        "_type": "mytype",
        "_id": "1",
        "_score": 1,
        "_source": {
          "field1": "Hello World!"
        }
      }
    ]
  }
}
```

**Comment by nik9000** [comment]
> "_source": 1

Nice!

**Comment by clintongormley** [comment]
This is the equivalent of _source: true.  _source: 0 is the equivalent of _source:false

**Comment by n0othing** [comment]
Binary switches like 0 and 1 make perfect sense for false/true, but anything > 1 will return all fields in the source (e.g.):

```
{
  "_source": 54684
}
```

**Comment by s1monw** [comment]
we have to deprecate the perl client first! I am +1 right here

**Comment by s1monw** [comment]
I think we should remove this ability slowly (deprecate in 5.x and remove in 6.x?) By this I mean the support for anything that `true|false` on boolean fields. I think our API should b NOT lenient her at all. Clients should fix that n their end

**Comment by s1monw** [comment]
@clintongormley can you pick this up as a task for deprecation

**Comment by javanna** [comment]
This was fixed by #22200.
